samples/test/util.py

```diff
@@ -101,15 +138,114 @@ def run_pipeline(
             )
             print("Run details page URL:")
             print(f"{external_host}/#/runs/details/{run_result.run_id}")
-            run_response = run_result.wait_for_run_completion(10 * MINUTE)
-            run = run_response.run
-            from pprint import pprint
-            # Hide detailed information
-            run.pipeline_spec.workflow_manifest = None
-            pprint(run)
-            return run
-
-        callback(run_pipeline=run_pipeline)
+            run_detail = run_result.wait_for_run_completion(10 * MINUTE)
+            # Hide detailed information for pretty printing
+            workflow_spec = run_detail.run.pipeline_spec.workflow_manifest
+            workflow_manifest = run_detail.pipeline_runtime.workflow_manifest
+            run_detail.run.pipeline_spec.workflow_manifest = None
+            run_detail.pipeline_runtime.workflow_manifest = None
+            pprint(run_detail)
+            # Restore workflow manifest, because test cases may use it
+            run_detail.run.pipeline_spec.workflow_manifest = workflow_spec
+            run_detail.pipeline_runtime.workflow_manifest = workflow_manifest
+            return run_detail
+
+        mlmd_connection_config = metadata_store_pb2.MetadataStoreClientConfig(
+            host=metadata_service_host,
+            port=metadata_service_port,
+        )
+        callback(
+            run_pipeline=run_pipeline,
+            mlmd_connection_config=mlmd_connection_config
+        )
 
     import fire
     fire.Fire(main)
+
+
+@dataclass
```

Comment:
FYI, this requires Python 3.7, but our min requirement is Py3.5:
https://github.com/kubeflow/pipelines/blob/82f9bfae347a84163ba174db9e4c1a98e9e80f47/sdk/python/setup.py#L108

It's not great for local testing, but I'm ok with this change if @chensun thinks we can update our min requirement to 3.7 sometime soon.

Reply:
We're dropping support for Python 3.5: https://github.com/kubeflow/pipelines/pull/4611
but I don't think we will drop Python 3.6 any time soon.

Reply:
This might be fine since e2e test infra does not need to have same requirements as the SDK.
This code will be executed in containers, so we control the image and the python version.

Reply:
Yes, it's only for testing.
Do you usually use 3.6 for local testing?